### PR TITLE
Add testing utilities to documents

### LINKED ISSUE #2880: Register a testing utility to the document
It is recommended for developers to use [`chainer.testing.get_trainer_with_mock_updater`](https://github.com/chainer/chainer/blob/master/chainer/testing/training.py#L8) to test trainers and related modules. But it is not registered to the document.

### PR DESCRIPTION
**Pull request by kmaehashi**
Took over @delta2323 -san's work (#4484). Fixes #2880.

### CODE REVIEW

File: docs/source/reference/check.rst [hunk 1]
```diff
@@ -45,13 +45,86 @@ and :class:`cupy.ndarray`.
...
+
+   chainer.testing.save_and_load
+   chainer.testing.save_and_load_hdf5
+   chainer.testing.save_and_load_npz
+   chainer.testing.get_trainer_with_mock_updater
+
+Trainer Extension Testing Utilities
+-----------------------------------
+
+Utilities for testing Trainer extensions.
```
Comment (asi1024) [inline docs/source/reference/check.rst]: Trainer -> trainer

Comment (kmaehashi) [reply docs/source/reference/check.rst]: Fixed.

**Comment by asi1024** [comment]
LGTM.

**Comment by asi1024** [comment]
jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**